### PR TITLE
Use chefstyle linting

### PR DESCRIPTION
**Pull request by sanjain-progress**
Signed-off-by: sachin007jain <sachin.jain@chef.io>

## Description
- Using Chefstyle for linting instead of Rubocop

## Related Issue
- https://github.com/chef/chef-workstation/issues/2522

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Chore (non-breaking change that does not add functionality or fix an issue)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have run the pre-merge tests locally and they pass.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] All commits have been signed-off for [the Developer Certificate of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: .github/workflows/linters.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+name: linters
+
+'on':
+  pull_request:
+  push:
+    branches:
+      - main
+
+jobs:
+  yamllint:
```
Comment (tas50) [inline .github/workflows/linters.yml:11]: Let's not do yaml lint here

Comment (sanjain-progress) [reply .github/workflows/linters.yml:11]: Should I remove the yaml lint job from here?
yeah, I see linters are failing.

Comment (tas50) [reply .github/workflows/linters.yml:11]: Yeah. That's not something we really care about internally

Comment (sanjain-progress) [reply .github/workflows/linters.yml]: Done

**Review by tas50 — APPROVED** [review]

**Pull request merged**